tests/test_unit_command.cc

```diff
@@ -18,9 +18,104 @@ namespace sc2 {
             test_ability_ = ABILITY_ID::ATTACK;
         }
 
+        void AdditionalTestSetup() {
+            agent_->Debug()->DebugCreateUnit(test_unit_type_, origin_pt_, agent_->Observation()->GetPlayerID(), 2);
+            target_point_ = GetPointOffsetX(origin_pt_);
+            agent_->Debug()->DebugCreateUnit(sc2::UNIT_TYPEID::ZERG_ZERGLING, GetPointOffsetX(origin_pt_, -10), 2, 1);
+            agent_->Debug()->SendDebug();
+        }
+
         void SetTestTime() override {
-            wait_game_loops_ = 50;
+            wait_game_loops_ = 150;
         }
+
+        void IssueUnitCommand(ActionInterface* act) {
+
+            if (test_tags_counter_ < TAGS_SUBTESTS_QUANTITY) {
+                test_units_tags_ = GetTagsFromUnits(test_units_);
+
+                if (!orders_verified_ && ability_command_sent_) {
+                    VerifyUnitOrders(test_unit_, test_ability_);
+                }
+
+                if (second_pass && ability_command_sent_) {
+                    orders_verified_ = false;
+                    ability_command_sent_ = false;
+                    first_pass = false;
+                    second_pass = false;
+                    test_tags_counter_++;
+                    return;
+                }
+
+                if (!second_pass && ability_command_sent_) {
+                    test_unit_tag_ = test_units_tags_.front();
+                    orders_verified_ = false;
+                }
+
+                uint32_t current_game_loop = agent_->Observation()->GetGameLoop();
+                order_on_game_loop_ = current_game_loop + 10;
+
+                switch (test_tags_counter_) {
+                case 0: TestTagsTargetingPoint(act);
```

Comment:
I think this should be separate test classes sharing the same setup code. The current implementation make the code much more complex (as you need to handle these first and second passes) while it is much better to keep the tests as simple as possible.